src/test/java/hudson/plugins/mercurial/HgExeFunctionalTest.java

```diff
@@ -120,4 +122,14 @@ public void credentialsUsernamePasswordTest() throws Exception {
         assertEquals("hg --config defaults.clone=--uncompressed clone http://some.thing/", b.toString());
     }
 
+    @Test public void checkVersion() throws Exception {
+        HgExe hgexe = new HgExe(
+                this.mercurialInstallation, null,
+                this.launcher, j.jenkins,
+                this.listener, this.vars);
+        String version = hgexe.version();
+        assertNotNull(version);
+        assertTrue(version.length() >= 3); // min: X.Y
+        assertTrue(version.length() <= 8); // max: XX.YY.ZZ
```

Comment:
IIRC it is possible for dev builds to have long version strings. I would just advise checking that `new VersionNumber(version).compareTo(new VersionNumber("1.0")) >= 0` to make sure it is parsable.